webapp/src/selectors/index.ts

```diff
@@ -15,15 +15,18 @@ export const getPluginServerRoute = (state) => {
     const config = getConfig(state);
 
     let basePath = '';
+    let finalPath = '';
+
     if (config && config.SiteURL) {
         basePath = new URL(config.SiteURL).pathname;
-
         if (basePath && basePath[basePath.length - 1] === '/') {
             basePath = basePath.substr(0, basePath.length - 1);
         }
-    }
+        const origin = new URL(config.SiteURL).origin;
 
-    return basePath + '/plugins/' + PluginId;
+        finalPath = `${origin}${basePath}/plugins/${PluginId}`;
+    }
+    return finalPath;
```

Comment:
This will not work when the site URL is not set, since `finalPath` is supposed to have the `plugins/(plugin id)` suffix, but it would be a blank string in this case. Although we generally check for the site URL to be defined on plugin startup, it's concerning that there is still a bug here, and potentially prone to other bugs if this code is modified later.

Reply:
@mickmister I see two options here 

- we fetch the site url from window.location.href and apply instead of config.SiteURL
- throw an error when the site url is not configured. 

I like the first option. what do you suggest

Reply:
[Done Michael. I have added a handler that checks the url is present or not ](https://github.com/mattermost/mattermost-plugin-jira/blob/4acda125c42617a33d78bf741bd958273e2d5b88/webapp/src/selectors/index.ts#L14-L31)

Reply:
We don't want to use the `window.locatiom.href`'s `pathname ` though. That will contain `/channels/(team)` etc.